tests/hypothesis/getitem_test.py

```diff
@@ -155,79 +155,23 @@ def test_getitem(
         )
     )
 
-    # IndexError: Offset must be non-negative (pyarrow does not support negative indexing)
-    assume(
-        not (
-            pandas_or_pyarrow_constructor is pyarrow_table_constructor
-            and isinstance(selector, slice)
-            and isinstance(selector.start, int)
-            and selector.start < 0
-        )
-    )
-    assume(
-        not (
-            pandas_or_pyarrow_constructor is pyarrow_table_constructor
-            and isinstance(selector, slice)
-            and isinstance(selector.stop, int)
-            and selector.stop < 0
-        )
-    )
-
-    # Pairs of slices are not supported
-    # NB a few trivial cases are supported, eg df[0:1, :]
-    # TypeError: Got unexpected argument type <class 'slice'> for compute function
+    # NotImplementedError: Slicing with step is not supported on PyArrow tables
     assume(
         not (
             pandas_or_pyarrow_constructor is pyarrow_table_constructor
             and isinstance(selector, tuple)
-            and isinstance(selector[0], slice)
-            and isinstance(selector[1], slice)
             and (
-                selector[0] != slice(None, None, None)
-                or selector[1] != slice(None, None, None)
+                (isinstance(selector[0], slice) and selector[0].step is not None)
+                or (isinstance(selector[1], slice) and selector[1].step is not None)
             )
         )
     )
-
-    # df[[], "a":], df[[], :] etc fail in pyarrow:
-    # ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (int64, null)
-    assume(
-        not (
-            pandas_or_pyarrow_constructor is pyarrow_table_constructor
-            and isinstance(selector, tuple)
-            and isinstance(selector[0], list)
-            and len(selector[0]) == 0
-            and isinstance(selector[1], slice)
-        )
-    )
-
-    # df[[], "a":], df[[], :] etc return different results between pandas/polars:
-    assume(
-        not (
-            pandas_or_pyarrow_constructor is pandas_constructor
-            and isinstance(selector, tuple)
-            and isinstance(selector[0], list)
-            and len(selector[0]) == 0
-            and isinstance(selector[1], slice)
-        )
-    )
-
-    # df[..., ::step] is not fine:
-    # TypeError: Expected slice of integers or strings, got: <class 'slice'>
-    assume(
-        not (
-            isinstance(selector, tuple)
-            and isinstance(selector[1], slice)
-            and selector[1].start is None
-            and selector[1].stop is None
-        )
-    )
```

Comment:
@sjdenny check this out, we can finally remove these 🥳 

Thank you _so_ much for having set up this hypothesis test, it was _so_ useful to finding edge cases 🙌 legend

Reply:
Amazing!  Glad it was useful :)